mitiq/tests/test_measurement_result.py

```diff
@@ -69,16 +71,69 @@ def test_empty():
     assert result.result == []
 
 
-def test_iter():
-    bitstrings = [[0, 1, 1], [0, 1, 0], [1, 0, 0]]
-
-    for m, bits in zip(MeasurementResult(bitstrings), bitstrings):
-        assert m == bits
-
-
 def test_convert_to_array():
     bitstrings = [[0, 0], [0, 1], [1, 0]]
     result = MeasurementResult(bitstrings)
     assert np.allclose(result.asarray, np.array(bitstrings))
 
     assert np.allclose(MeasurementResult([]).asarray, np.array([]))
+
+
+@pytest.mark.parametrize("qubit_indices", ((0, 1), (1, 20)))
+def test_measurement_result_with_strings(qubit_indices):
+    """Try using strings instead of lists of integers."""
+    bitstrings = ["00", "01", "10"]
+    int_bitstrings = [[0, 0], [0, 1], [1, 0]]
+
+    result = MeasurementResult(bitstrings, qubit_indices=qubit_indices)
+
+    assert result.nqubits == 2
+    assert result.qubit_indices == qubit_indices
+    assert result.shots == 3
+    assert np.allclose(result.result, int_bitstrings)
+
+
+@pytest.mark.parametrize("qubit_indices", ((0, 1), (1, 20)))
+def test_measurement_result_from_counts(qubit_indices):
+    """Test initialization from a dictionary of counts."""
+    counts = {"00": 1, "01": 2}
+    int_bitstrings = [[0, 0], [0, 1], [0, 1]]
+
+    result = MeasurementResult.from_counts(
+        counts=counts,
+        qubit_indices=qubit_indices,
+    )
+    assert result.nqubits == 2
+    assert result.qubit_indices == qubit_indices
+    assert result.shots == 3
+    assert np.allclose(result.result, int_bitstrings)
+
+
+def test_measurement_result_get_counts():
+    """Test initialization from a dictionary of counts."""
+    counts = {"00": 1, "01": 2}
+
+    int_bitstrings = [[0, 0], [0, 1], [0, 1]]
+    result = MeasurementResult(
+        result=int_bitstrings,
+        qubit_indices=(1, 20),
+    )
+    assert result.get_counts() == counts
+    # Info about qubit indices is expected to be lost
+    new_res = MeasurementResult.from_counts(result.get_counts())
+    assert new_res.qubit_indices != result.qubit_indices
```

Comment:
The qubit indices should not only not be equal, but it should be None, right?
```suggestion
    assert new_res.qubit_indices is None
```

Reply:
Changed to `assert new_res.qubit_indices = (0, 1)` since qubit_indeces are initialized to default values in `__post_init__`.